pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+    )
+    assert_series_equal(move_df.count(), expected)
+
+
+def test_group_by():
+    move_df = _default_move_df()
+
+    expected = _default_pandas_df()
+    expected = expected.groupby('id').mean()
+    assert_frame_equal(move_df.groupby(TRAJ_ID).mean(), expected)
+
+
+def test_select_dtypes():
+    move_df = _default_move_df()
+
+    expected = DataFrame(data=[
+        [39.984094, 116.319236],
+        [39.984198, 116.319322],
+        [39.984224, 116.319402],
+        [39.984224, 116.319402],
+    ], columns=['lat', 'lon'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(move_df.select_dtypes(include='float64'), expected)
+
+
+def test_astype():
+    move_df = _default_move_df()
+
+    expected = DataFrame(data=[
+        [39, 116, 1224741185000000000, 1],
+        [39, 116, 1224741186000000000, 1],
+        [39, 116, 1224741191000000000, 2],
+        [39, 116, 1224741191000000000, 2],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(move_df.astype('int'), expected)
+
+
+def test_sort_values():
+    move_df = _default_move_df()
+
+    expected = DataFrame(data=[
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[2, 3, 0, 1]
+    )
+
+    assert_frame_equal(move_df.sort_values(by=TRAJ_ID, ascending=False), expected)
+
+
+def test_reset_index():
+    move_df = _default_move_df()
+
+    move_df = move_df.loc[1:]
+    assert_array_equal(move_df.index, [1, 2, 3])
+    move_df.reset_index(inplace=True)
+    assert_array_equal(move_df.index, [0, 1, 2])
+
+
+def test_set_index():
+    move_df = _default_move_df()
+
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05')],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06')],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11')],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11')],
+    ], columns=['lat', 'lon', 'datetime'], index=[1, 1, 2, 2]
+    )
+    expected.index.name = 'id'
+    assert_frame_equal(move_df.set_index('id'), expected)
+
+    try:
+        move_df.set_index('datetime', inplace=True)
+        assert False
+    except AttributeError:
+        assert True
+
+
+def test_drop():
+    move_df = _default_move_df()
+    move_df[UID] = [1, 1, 2, 3]
+
+    move_test = move_df.drop(columns=[UID])
+    assert UID not in move_test
+    assert UID in move_df
+    assert isinstance(move_test, PandasMoveDataFrame)
+
+    move_test = move_df.drop(index=[0, 1])
+    assert move_test.len() == 2
+    assert isinstance(move_test, PandasMoveDataFrame)
+
+    move_df.drop(columns=[UID], inplace=True)
+    assert UID not in move_df
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574968)